### PR TITLE
Add vsphere volume operations (#586)

### PR DESCRIPTION
**Pull request by fabioDMFerreira**
* Power on a vsphere instance after creating it.

TODO:
* Deleting vsphere volume
* Using vsphere object manager on volume operations

### CODE REVIEW

File: vsphere/vsphere_volume.go [hunk 1]
```diff
@@ -1,29 +1,295 @@
...
+	if err != nil {
+		return vol, err
+	}
+
+	// TODO: Implement an alternative solution to register the disk using the object manager.
+	// The advantages are the simplification of operations like getting attachments and deleting
+	// I could not having this working due to the error stated in https://github.com/vmware/govmomi/issues/2174.
+	// objectManager := vslm.NewObjectManager(ds.Client())
+
+	// _, err = objectManager.RegisterDisk(context.TODO(), ds.NewURL("volumes/"+vol.Name+".vmdk").String(), vol.Name)
```
Comment (eyberg) [inline vsphere/vsphere_volume.go:63]: can we cut?

File: vsphere/vsphere_volume.go [hunk 2]
```diff
@@ -1,29 +1,295 @@
...
+
+	for _, file := range files {
+		path := file.GetFileInfo().Path
+		if !strings.Contains(path, "-flat.vmdk") {
+			*vols = append(*vols, lepton.NanosVolume{Name: path})
+		}
+	}
+
+	// TODO: use object manager to get all volumes listed and convert them to nanos volumes, blocked by https://github.com/vmware/govmomi/issues/2174
+	// disks, err := v.getAllVolumes(ds)
```
Comment (eyberg) [inline vsphere/vsphere_volume.go:135]: minus the todo; can we cut?

File: vsphere/vsphere_volume.go [hunk 3]
```diff
@@ -1,29 +1,295 @@
...
+	// }
+
+	return vols, nil
 }
 
 // DeleteVolume is a stub to satisfy VolumeService interface
 func (v *Vsphere) DeleteVolume(ctx *lepton.Context, name string) error {
-	return nil
+	// TODO: Blocked by https://github.com/vmware/govmomi/issues/2174
+	// f := find.NewFinder(v.client, true)
```
Comment (eyberg) [inline vsphere/vsphere_volume.go:146]: this chunk of code should be rm'd

**Review by eyberg — APPROVED** [review]
just need to get rid of all the commented out code

**Comment by fabioDMFerreira** [comment]
Once we overpass the problem in https://github.com/vmware/govmomi/issues/2174, we can uncomment the code and remove the current implementation.

The commented code is the code using the alternative solution. Basically, the code just need to be tested.

**Comment by eyberg** [comment]
I would either leave the code in or get rid of it

**Comment by fabioDMFerreira** [comment]
In that case, I'll opt to uncomment the code and delete the current implementation. It's unlikely it will work at the first attempt, but it's a cleaner solution.

**Pull request merged**